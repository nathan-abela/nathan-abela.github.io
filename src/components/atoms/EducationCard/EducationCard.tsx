import { FC } from 'react';
import { format } from 'date-fns';

import { IEducation } from '@Types';
import { Flex, StyledEducation } from './styles';

export interface EducationProps {
	education: IEducation;
}

const EducationCard: FC<EducationProps> = ({ education }) => {
	return (
		<StyledEducation>
			<Flex>
				<div>
					<strong>
						{education.website ? (
							// eslint-disable-next-line prettier/prettier
							<a
								href={education.website}
								target="_blank"
								rel="noopener noreferrer"
							>
								{education.course}
							</a>
						) : (
							education.course
						)}
					</strong>
				</div>

				<div>
					{format(new Date(education.startDate), 'yyyy')} - {format(new Date(education.endDate), 'yyyy')}
				</div>
			</Flex>

			<strong>
				<p>
					{education.institute} | {education.location}
				</p>
			</strong>

			<p>{education.description}</p>

			{education.bullets.length > 0 && (
				<ul style={{ paddingLeft: 16 }}>
					{education.bullets.map((bullet, j) => (
						<li key={j}> &bull; {bullet} </li>
					))}
				</ul>
			)}

			{education.technologies.length > 0 && (
				<small>
					<i>
						<strong>Technologies:</strong> {education.technologies.join(', ')}
					</i>
				</small>
			)}
		</StyledEducation>
	);
};

export { EducationCard };
