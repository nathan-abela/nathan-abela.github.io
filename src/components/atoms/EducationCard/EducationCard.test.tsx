import { render } from '@testing-library/react';

import { EducationCard } from './EducationCard';

describe('Education Card', () => {
	it('renders education card correctly', () => {
		const { container } = render(
			<EducationCard
				education={{
					institute: 'Malta College for Arts, Science and Technology (MCAST)',
					website: 'https://mcast.edu.mt',
					location: 'Paola, Malta',
					course: 'Bachelor of Science (Honours) in Software Development',
					startDate: '2021',
					endDate: '2024',
					description:
						'Focuses on creating software applications using formal design patterns, and solving complex real-world problems using formal methodologies. Includes units such as secure software development, server-side and client-side scripting, discrete mathematics, and other similar modules.',
					bullets: [],
					technologies: [
						'JavaScript',
						'Angular',
						'Bootstrap',
						'Java',
						'Selenium',
						'RESTful API',
						'C#',
						'ASP.NET Core',
						'Android Studio',
						'Firebase',
						'Amazon Web Services',
						'Algorithms',
						'Data Structures',
						'OWASP ZAP',
						'C',
						'Python',
						'Tensorflow',
						'LaTeX',
						'SQL',
						'SQL Data Modeller',
						'SQL Server Management Studio',
						'Power BI',
						'CakePHP',
						'PHP',
						'XAMPP',
					],
				}}
			></EducationCard>
		);
		expect(container).toMatchSnapshot();
	});
});
