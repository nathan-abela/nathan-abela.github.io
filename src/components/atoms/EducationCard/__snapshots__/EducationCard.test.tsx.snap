// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Education Card renders education card correctly 1`] = `
<div>
  <div
    class="styles__StyledEducation-sc-1mmyy45-0 kEasSY"
  >
    <div
      class="styles__Flex-sc-1mmyy45-1 cFtOTd"
    >
      <div>
        <strong>
          <a
            href="https://mcast.edu.mt"
            rel="noopener noreferrer"
            target="_blank"
          >
            Bachelor of Science (Honours) in Software Development
          </a>
        </strong>
      </div>
      <div>
        2021
         - 
        2024
      </div>
    </div>
    <strong>
      <p>
        Malta College for Arts, Science and Technology (MCAST)
         | 
        Paola, Malta
      </p>
    </strong>
    <p>
      Focuses on creating software applications using formal design patterns, and solving complex real-world problems using formal methodologies. Includes units such as secure software development, server-side and client-side scripting, discrete mathematics, and other similar modules.
    </p>
    <small>
      <i>
        <strong>
          Technologies:
        </strong>
         
        JavaScript, Angular, Bootstrap, Java, Selenium, RESTful API, C#, ASP.NET Core, Android Studio, Firebase, Amazon Web Services, Algorithms, Data Structures, OWASP ZAP, C, Python, Tensorflow, LaTeX, SQL, SQL Data Modeller, SQL Server Management Studio, Power BI, CakePHP, PHP, XAMPP
      </i>
    </small>
  </div>
</div>
`;
